### PR TITLE
Add new-style api reference docs for 1.5

### PR DESCRIPTION
**Pull request by pwittrock**
**Note: This will be hard to review the source by hand because everything checked in is generated from other files.  I can commit the source for the generated files in a separate PR.**

Changes here: https://deploy-preview-1818--kubernetes-io-vnext-staging.netlify.com/docs/reference/
See the "New Beta Docs" links in the Table of Contents

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1818)

### CODE REVIEW

**Comment by pwittrock** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1818)

File: docs/api-reference/1_5/documents/_overview.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# <strong>API OVERVIEW</strong>
+
+Welcome to the Kubernetes API.  You can use our API to access Kubernetes API endpoints to read and write Kubernetes
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: "You can use the Kubernetes API to read and write Kubernetes resource objects via a Kubernetes API endpoint."

File: docs/api-reference/1_5/documents/_overview.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+resource objects.
+
+## Resource Categories
+
+This is a highlevel overview of the basic types of resources provide by the Kubernetes API and their primary functions.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: highlevel -> high-level

File: docs/api-reference/1_5/documents/_overview.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+**Workloads** *Run your containers*: These are responsible for managing and running your containers on the cluster.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: **Workloads** are objects you use to manage and run your containers on the cluster.

File: docs/api-reference/1_5/documents/_overview.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+**Discovery & LB Resources** *Access your containers*: These are responsible for stitching your workloads together into an externally accessible Loadbalanced Service.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: **Discovery and Load-Balancing** resources are objects you use to "stitch" your workloads together into an externally accessible, load-balanced Service.

File: docs/api-reference/1_5/documents/_overview.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+**Config & Storage Resources** *Inject initialization data into your containers*: These are responsible for injecting data into your applications and persisting data externally to your container.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: **Configuration and Storage** resources are objects you use to inject initialization data into your applications, and to persist data that is external to your container.

File: docs/api-reference/1_5/documents/_overview.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+**Cluster Resources** *Manage your cluster*: These are responsible for defining configuration of the cluster itself, and are generally only used by cluster operators.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: **Cluster Resource** objects define how the cluster itself is configured; these are typically used only by cluster operators.

File: docs/api-reference/1_5/documents/_overview.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+**Metadata Resources** *Configure resource meta behavior*: These are responsible for configuring behavior of your other Resources within the Cluster.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_overview.md]: **Metadata** resources are objects you use to configure the behavior of other resources within the cluster, such as...  (Fill in "such as" here)

File: docs/api-reference/1_5/documents/_overview.md (hunk 1 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 2 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 3 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 4 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 5 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 6 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_overview.md (hunk 7 above)
Comment (pwittrock) [inline docs/api-reference/1_5/documents/_overview.md]: Done

File: docs/api-reference/1_5/documents/_cluster.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# <strong>CLUSTER</strong>
+
+These are responsible for defining configuration of the cluster itself, and are generally only used by cluster operators.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_cluster.md]: "Cluster Resources are responsible for..."

File: docs/api-reference/1_5/documents/_config.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# <strong>CONFIG & STORAGE</strong>
+
+These are responsible for injecting data into your applications and persisting data externally to your container.
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_config.md]: "Configuration and Storage resources are responsible for..."

File: docs/api-reference/1_5/documents/_definitions.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+# <strong>DEFINITIONS</strong>
+
+These are Definitions for objects used by the above APIs
```
Comment (devin-donnelly) [inline docs/api-reference/1_5/documents/_definitions.md]: "This section contains definitions for objects used in the Kubernetes APIs."

**Comment by devin-donnelly** [comment]
In general, replace the "These are" constructs in each resource category header as I've specified in the comments, and this is fine for me.

**Comment by devin-donnelly** [comment]
Also, the "Getting Started" section of the kubectl command reference should probably add a paragraph about how "run" and "expose" are the most basic commands to get a workload running on your cluster and exposed to the outside world to serve requests. WDYT?

**Comment by pwittrock** [comment]
> Also, the "Getting Started" section of the kubectl command reference should probably add a paragraph about how "run" and "expose" are the most basic commands to get a workload running on your cluster and exposed to the outside world to serve requests. WDYT?

Yeah, each of the sections could probably use a brief summary.  I can add that.

**Comment by pwittrock** [comment]
Alright, I think that I addressed you comments.  I added some sections for the categories in the kubectl ref section.  I also added a section without the operations since Brian was asking for it.  It is the same as the api, but just missing the operations.

**Comment by devin-donnelly** [comment]
Okay, LGTM.

**Comment by pwittrock** [comment]
Woot.

**Pull request merged**